sdk/src/client/secret/mod.rs

```diff
@@ -587,12 +587,12 @@ where
 
     validate_signed_transaction_payload_length(&tx_payload)?;
 
-    let conflict = verify_semantic(&inputs_data, &tx_payload)?;
-
-    if let Some(conflict) = conflict {
-        log::debug!("[sign_transaction] conflict: {conflict:?} for {:#?}", tx_payload);
-        return Err(Error::TransactionSemantic(conflict));
-    }
+    // TODO: move validation to another place or only validate signatures here
```

Comment:
Even to validate signatures we need the protocol parameters. But leave it as TODO anyway, I already added protocol parameters to the trait in my PR.

Reply:
Why do we need the protocol parameters for the signatures validation?

Reply:
We need min/max commitable age to determine the address (if expired or not) 😭

Reply:
Ah for inputs with expiration unlock condition I guess